Documentation/tls/tls-etcd.md

```diff
@@ -0,0 +1,90 @@
+
+## Usage
+
+Comment out the existing signed etcd TLS in your platform, i.e. `platforms/aws/tectonic.tf`:
+
+```
+/*
+module "etcd_certs" {
+  source = "../../modules/tls/etcd"
+
+  etcd_ca_cert_path     = "${var.tectonic_etcd_ca_cert_path}"
+  etcd_cert_dns_names   = "${data.template_file.etcd_hostname_list.*.rendered}"
+  etcd_client_cert_path = "${var.tectonic_etcd_client_cert_path}"
+  etcd_client_key_path  = "${var.tectonic_etcd_client_key_path}"
+  self_signed           = "${var.tectonic_etcd_tls_enabled}"
+  service_cidr          = "${var.tectonic_service_cidr}"
+}
+*/
+```
+
+Configure the user provided certificate paths in your platform, i.e. `platforms/aws/tectonic.tf`:
+
+```
+module "etcd_certs" {
+  source = "../../modules/tls/etcd/user-provided"
+
+  etcd_ca_crt_pem_path     = "/path/to/etcd-ca.crt"
+  etcd_server_crt_pem_path = "/path/to/etcd-server.crt"
+  etcd_server_key_pem_path = "/path/to/etcd-server.key"
+  etcd_peer_crt_pem_path   = "/path/to/etcd-peer.crt"
+  etcd_peer_key_pem_path   = "/path/to/etcd-peer.key"
+  etcd_client_crt_pem_path = "/path/to/etcd-client.crt"
+  etcd_client_key_pem_path = "/path/to/etcd-client.key"
+}
+```
+
+The signed etcd server certificate must have the following Subject Alternative Name (SAN) and Key Usage associations:
+
+```
+$ openssl x509 -noout -text -in /path/to/etcd-server.crt
+Certificate:
+...
+        X509v3 extensions:
+            X509v3 Key Usage: critical
+                Key Encipherment
+            X509v3 Extended Key Usage:
+                TLS Web Server Authentication
+...
+            X509v3 Subject Alternative Name:
+                DNS:<tectonic_cluster_name>-etcd-0.<tectonic_base_domain>,
+                DNS:<tectonic_cluster_name>-etcd-1.<tectonic_base_domain>,
+                ...
+                DNS:<tectonic_cluster_name>-etcd-<tectonic_etcd_count-1>.<tectonic_base_domain>,
+```
+
+The signed etcd peer certificate must have the following Subject Alternative Name (SAN) and Key Usage associations:
+
+```
+$ openssl x509 -noout -text -in /path/to/etcd-peer.crt
+Certificate:
+        X509v3 extensions:
+            X509v3 Key Usage: critical
+                Key Encipherment
+            X509v3 Extended Key Usage:
+                TLS Web Server Authentication, TLS Web Client Authentication
+...
+            X509v3 Subject Alternative Name:
+                DNS:<tectonic_cluster_name>-etcd-0.<tectonic_base_domain>,
+                DNS:<tectonic_cluster_name>-etcd-1.<tectonic_base_domain>,
+                ...
+                DNS:<tectonic_cluster_name>-etcd-<tectonic_etcd_count-1>.<tectonic_base_domain>,
+```
+
+The signed etcd client certificate must have the following Key Usage associations:
+
+```
+$ openssl x509 -noout -text -in /path/to/etcd-client.crt
+Certificate:
+...
+        X509v3 extensions:
+            X509v3 Extended Key Usage:
+                TLS Web Client Authentication
+```
+
```

Comment:
Add a summary to each, just to confirm that you can go back to the main guide for booting the cluster.

> Now you’re ready to boot the cluster with Terraform using these certs…

Reply:
@robszumski 
so this is kinda a bug. The CA certificate for Peer and Client should be specified separately that we conflate them here is a problem.

We should arguably also cover auto-tls that etcd supports here.

Reply:
weird that I cannot reply below for the tls-etcd.md comment.

so, separate the client and peer certs into their own sections, and add a note that etcd supports auto-tls?

Reply:
I'd like @robszumski to weigh in here.
Perhaps we leave it be but it's kinda buggy in the way it's configured here.

Reply:
for now let's doc it as is

Reply:
I'll open bugs on the problem I have found